packages/network/src/helpers/MetricsContext.ts

```diff
@@ -26,6 +26,8 @@ export class Metrics {
     }
     private readonly recordedMetrics: {
         [key: string]: {
+            // eslint-disable-next-line no-underscore-dangle
+            _speedometer: Speedometer
```

Comment:
It would make the state unnecessary complicated, maybe not worth it. This is just a private data structure which is not accessible outside of `MetricsContext` class.